src/main/java/io/cryostat/net/web/http/api/v2/CertificatePostHandler.java

```diff
@@ -151,16 +151,18 @@ public boolean isOrdered() {
             throw new ApiException(409, filePath.toString() + " Certificate already exists");
         }
 
-        try (InputStream fis = fs.newInputStream(certPath);
-                FileOutputStream out = outputStreamFunction.apply(filePath.toFile())) {
-
+        try (InputStream fis = fs.newInputStream(certPath)) {
             Collection<? extends Certificate> certificates = certValidator.parseCertificates(fis);
```

Comment:
Looks pretty good now. Is it possible that `parseCertificates` throws no Exception, but returns an empty Collection? If so then I think it would make sense to exit early from here as well, to again avoid creating the output file. If it is not possible for an empty Collection to be returned then perhaps just leave a comment here explaining this.